pkg/controllers/clusters/clusterlifecycle/clusterlifecycle_controller.go

```diff
@@ -231,7 +234,7 @@ func (c *Controller) updateClusterTaints(ctx context.Context, cluster *clusterap
 func patchClusterTaints(ctx context.Context, client clusternetclientset.Interface, cluster *clusterapi.ManagedCluster,
 	taintsToAdd, taintsToRemove []*corev1.Taint) error {
 	var err error
-	var newCluster *clusterapi.ManagedCluster
+	var newCluster *clusterapi.ManagedCluster = cluster
```

Comment:
This is not needed. And we'd better not do this, since `cluster` is a pointer, we may modify `newCluster` inadvertently.

Reply:
Actually, at least one of `RemoveTaint` and `AddOrUpdateTaint` will be performed. https://github.com/clusternet/clusternet/blob/main/pkg/controllers/clusters/clusterlifecycle/clusterlifecycle_controller.go#L222-L225

Reply:
yes, if RemoveTaint isn't be performed,  the newCluster is nil, and AddOrUpdateTaint will recieve a nil, then a panic happens in AddOrUpdateTaint

Reply:
Ok, got it. You're right. Nice catch.

Let's use 
```go
newCluster := cluster.DeepCopy()
```

Reply:
There is a cluster.DeepCopy() in AddOrUpdateTaint